### ISSUE TITLE
Some troubles with specification

### ISSUE THREAD

**Issue by Anyeos** [AppImage/AppImageKit#955]
Sorry guys but it is some confusing and time consuming for me being so "free" in the specifications.

I suggest a review with more practical cases.

Because in my case I need to maintain six machines. And with five different flavours (different person):
1) My father likes to play games, and he need some apps to surf the web, maybe take notes, and maybe handle a few more things.
2) My mother play games in FB, and handle her pictures. She need always a working Picture Viewer, or manager, or photo album or something like that.
3) My sister too play games but she write documents so she need software to write documents and sometimes she handles pictures.
4) A friend compose music and do a few more things. He too make drawings, he is a drawer so he need specialiced updated drawing software like Krita.
5) Me. I am a developer. I too work with multimedia and music like my friend and some graphical jobs. 

The real case is when I need to update all that systems to get updated apps. The problem is that I need to do a lot of work just only installing an AppImage because:
1) It does not includes an icon by default. So I must copy and install one from somewhere.
2) It does not include a .desktop file neither. I must create or install and edit one.
3) No everybody wants to click over an app with a icon like a striked paper (default no-icon in file managers like nautilus). So it is some uncomfortable looking for the apps.

The solution can be to make including an icon a requisite.
We can develop something like (exe-thumbnailer) so Nautilus and likewise can show the real app icon extracted from the AppImage.

Why I think that. Because if my sister, my mother, my father,... need to install something. It will be more easy if it is more like a Windows exe showing an icon (that helps a lot identifying a software). So, they will don't need me to update the apps.

And, if I must install something like AppImageLauncher... Where is the point of having an AppImage? The only advantage is more projects are packaging the software in AppImage format. So I can get an updated binary easily but installing something like AppImageLauncher add some uneeded complexity.

So, as I said above, the solution can be impose the use of an icon and make an "appimage-thumbnailer" plugin for file managers like Nautilus.

I hope you understand me. I don't want to impose nothing. I just want to present my real case of use.

**Comment by probonopd** [comment]
Hi @Anyeos and welcome to AppImage. Thanks for sharing your thoughts. This sort of feedback is very valuable to us.

> The solution can be to make including an icon a requisite.

An icon is mandatory as per [the AppImage specification](https://github.com/AppImage/AppImageSpec/blob/master/draft.md#the-filesystem-image):

> **MUST** contain a .DirIcon file as per the AppDir specification

The .DirIcon is used by thumbnailers to render the icon onto the AppImage file in the file manager.

Unfortunately, some desktops cannot use this to render the icon into the menus, so if you want that, you also need icon files below `usr/share/icons/hicolor` following the Icon Theme Specification for the icon identifier as set in the `Icon= key` of the `$APPNAME.desktop` file. This icon or set of icons will be used by desktop integration tools like the optional [`appimaged`](https://github.com/AppImage/appimaged/) daemon (or the third-party AppImageLauncher) for the menus.

> **MAY** contain icon files below usr/share/icons/hicolor following the Icon Theme Specification for the icon identifier as set in the Icon= key of the $APPNAME.desktop file. If present, these icon files SHOULD be given preference as the icon being used to represent the AppImage.

Given your comment, I will change this to **SHOULD** in the spec. We have been recommending it to our uses anyway, and our tools have been doing it since a long time.

> It does not include a .desktop file neither. I must create or install and edit one.

As per [the spec](https://github.com/AppImage/AppImageSpec/blob/master/draft.md#the-filesystem-image), it is not strictly required, but in practice most AppImages do contain it:

> **SHOULD** contain exactly one `$APPNAME.desktop` file in its root directory with `$APPNAME` being the name of the payload application

> We can develop something like (exe-thumbnailer) so Nautilus and likewise can show the real app icon extracted from the AppImage.

Err, it exists. This is how it looks with `appimaged` installed:

![icon](https://user-images.githubusercontent.com/2480569/57191348-cff5a980-6f24-11e9-8222-b2b1f61238cc.png)

But there is currently a bug https://github.com/AppImage/libappimage/issues/103 that prevents thumbails from being generated properly.

There is also a [GNOME thumbnailer](https://github.com/azubieta/appimage-gnome-thumbnailer) and a [KDE thumbnailer](https://github.com/KDE/kio-extras/commit/e88a4c3d4a0233e33333f66236e69700719f6290), but 

> And, if I must install something like AppImageLauncher... Where is the point of having an AppImage?

thumbnailers need to be installed in the system, too - unless they are pre-installed by distributions by default. (Distributions are pushing their own formats like Snappy and Flatpak thoguh.)

> And, if I must install something like AppImageLauncher... Where is the point of having an AppImage? 

Destop integration is entirely optional. You don't need it. You can run an AppImage like this:
https://discourse.appimage.org/t/how-to-run-an-appimage/80

**Comment by Anyeos** [comment]
Yes, you're right but I point from the perspective of practical use. I mean, thumb is something that will help. But when I was talking about AppImageLauncher I am comparing the simplicity of directly copy and run an AppImage. Something that is not very hard to learn for someone. Versus a more sofisticated thing like AppImageLauncher but if somebody wants to use it, there is.

It is a great new that there already exists a thumbnailer. But sad to ear that it is not working because there are a bug. Anyway, somebody will fix it some day, I hope.

I don't want more from AppImage, because it already do the right work. But I just wanted to tell some points to not forget the practical perspective.

So, what we need now is making work the thumbnailer and a matter to include the icon on the menus. But I think that the menu thing (the whole .desktop thing as freedesktop specs) got out of the AppImage purpose. Because I think it is ok to not having a menu icon, because the menu must be created by some installer and not by the Application itself. But that is my opinion.

**Comment by probonopd** [comment]
> It is a great new that there already exists a thumbnailer. But sad to ear that it is not working because there are a bug

Should be fixed in https://github.com/AppImage/appimaged/pull/78.
